pkg/odo/cli/component/list.go

```diff
@@ -64,6 +64,10 @@ func (lo *ListOptions) Validate() (err error) {
 
 	var project, app string
 
+	if len(lo.Application) != 0 && lo.allAppsFlag {
+		return odoutil.ThrowContextError()
```

Comment:
I would prefer an error like "--app and --all-apps cannot be provided together" or something more UX enhancing